### PR TITLE
Add tiflash execution tuning materials

### PR DESCRIPTION
**Pull request by yibin87**
Signed-off-by: yibin <huyibin@pingcap.com>

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tune-tiflash-performance.md [hunk 1]
```diff
@@ -13,25 +13,17 @@ aliases: ['/docs-cn/dev/tiflash/tune-tiflash-performance/','/docs-cn/dev/referen
 
 ## TiDB 相关参数调优
 
-1. 对于 OLAP/TiFlash 专属的 TiDB 节点，建议调大读取并发数 [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) 到 80：
+1. 当下推未生效时，可以尝试开启强制下推功能：
```
Comment (windtalker) [inline tiflash/tune-tiflash-performance.md:16]: 建议改成“当没有生成 mpp plan 的时候，可以尝试强制开启 mpp”

Comment (yibin87) [reply tiflash/tune-tiflash-performance.md]: Done

**Review by windtalker — APPROVED** [review]
LGTM

File: tiflash/tune-tiflash-performance.md [hunk 2]
```diff
@@ -61,4 +53,37 @@ aliases: ['/docs-cn/dev/tiflash/tune-tiflash-performance/','/docs-cn/dev/referen
...
+5. 尝试使用 Broadcast Hash Join 来代替 Shuffled Hash Join：
+
+    - [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为 bytes。如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+    - [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    set @@tidb_broadcast_join_threshold_count = 100000;
+    ```
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:67]:
```suggestion
        ```sql
        set @@tidb_broadcast_join_threshold_count = 100000;
        ```
```

File: tiflash/tune-tiflash-performance.md [hunk 3]
```diff
@@ -61,4 +53,37 @@ aliases: ['/docs-cn/dev/tiflash/tune-tiflash-performance/','/docs-cn/dev/referen
...
+
+6. 尝试设置更大的执行并发度：
+
+    [`tidb_max_tiflash_threads`](/system-variables.md#tidb_max_tiflash_threads-从-v610-版本开始引入)，单位为 bytes。用来设置 TiFlash 中 request 执行的最大并发度果表大小。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:71]:
```suggestion
    [`tidb_max_tiflash_threads`](/system-variables.md#tidb_max_tiflash_threads-从-v610-版本开始引入)，单位为 bytes。用来设置 TiFlash 中 request 执行的最大并发度。
```

File: tiflash/tune-tiflash-performance.md [hunk 4]
```diff
@@ -61,4 +53,37 @@ aliases: ['/docs-cn/dev/tiflash/tune-tiflash-performance/','/docs-cn/dev/referen
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:75]:
```suggestion
    ```sql
```

File: tiflash/tune-tiflash-performance.md [hunk 5]
```diff
@@ -61,4 +53,37 @@ aliases: ['/docs-cn/dev/tiflash/tune-tiflash-performance/','/docs-cn/dev/referen
...
+    set @@tidb_max_tiflash_threads = 20;
+    ```
+
+7. 尝试设置细粒度 Shuffle 的相关参数：
+
+    - [`tiflash_fine_grained_shuffle_stream_count`](/system-variables.md#tiflash_fine_grained_shuffle_stream_count-从-v620-版本开始引入)，单位为线程数。当窗口函数下推到 TiFlash 执行时，可以通过该变量控制窗口函数执行的并行度。
+    - [`tiflash_fine_grained_shuffle_batch_size`](/system-variables.md#tiflash_fine_grained_shuffle_batch_size-从-v620-版本开始引入)，单位为 bytes。细粒度 shuffle 功能开启时，下推到 TiFlash 的窗口函数可以并行执行。该变量控制发送端发送数据的攒批大小。
+
+    {{< copyable "sql" >}}
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:84]:
```suggestion
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: tiflash/tune-tiflash-performance.md [hunk 6]
```diff
@@ -1,64 +1,94 @@
...
-    ALTER TABLE employees COMPACT TIFLASH REPLICA;
+    set @@tidb_broadcast_join_threshold_count = 100000;
     ```
 
-    ```sql
-    ALTER TABLE employees COMPACT PARTITION pNorth, pEast TIFLASH REPLICA;
-    ```
\ No newline at end of file
+### 设置更大的执行并发度
+
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:79]: 感觉需要说明为什么设置更大的并发度会帮助调优性能

File: tiflash/tune-tiflash-performance.md [hunk 7]
```diff
@@ -1,64 +1,94 @@
...
+[`tidb_max_tiflash_threads`](/system-variables.md#tidb_max_tiflash_threads-从-v610-版本开始引入)，单位为 bytes。用来设置 TiFlash 中 request 执行的最大并发度。
+
+```sql
+set @@tidb_max_tiflash_threads = 20;
+```
+
+### 设置细粒度 Shuffle 参数
+
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:87]: 需要说明设置细粒度 shuffle 参数对性能调优的帮助。

File: tiflash/tune-tiflash-performance.md [hunk 8]
```diff
@@ -1,64 +1,94 @@
...
+ALTER TABLE employees COMPACT TIFLASH REPLICA;
+```
 
-    [`ALTER TABLE ... COMPACT`](/sql-statements/sql-statement-alter-table-compact.md) 可以触发 TiFlash 节点对某个表或者某个分区进行数据整理。数据整理时，表中的物理数据会被重写，如清理已删除的数据、合并多版本数据等，从而可以获得更高的访问性能，并减少磁盘空间占用。
+```sql
+ALTER TABLE employees COMPACT PARTITION pNorth, pEast TIFLASH REPLICA;
+```
+
+### 使用 Broadcast Hash Join 代替 Shuffled Hash Join
+
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:70]: 需要说明用 Broadcast Hash Join 代替 Shuffled Hash Join 对 tiflash 性能调优的好处

File: tiflash/tune-tiflash-performance.md [hunk 9]
```diff
@@ -1,64 +1,94 @@
...
+set @@tidb_opt_agg_push_down = 1;
+```
 
-    ```sql
-    set @@tidb_opt_agg_push_down = 1;
-    ```
+### 开启 `Distinct`
 
-4. 尝试开启 `Distinct` 推过 `Join` / `Union` 等 TiDB 算子的优化：
+可以开启 `Distinct` 推过 `Join` / `Union` 等 TiDB 算子的优化：
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:49]: 需要说明开启 `Distinct` 对 tilfash 性能调优的好处

File: tiflash/tune-tiflash-performance.md [hunk 10]
```diff
@@ -1,64 +1,94 @@
...
+- [设置更大的执行并发度](#设置更大的执行并发度)
+- [设置细粒度 Shuffle 参数](#设置细粒度-shuffle-参数)
 
-    ```sql
-    set @@tidb_distsql_scan_concurrency = 80;
-    ```
+### 强制开启 MPP 模式
 
-2. 开启 Super batch 功能：
+当没有生成 MPP 执行计划的时候，可以尝试强制开启 MPP：
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:29]: 需要说明开启 mpp 模式对性能调优的帮助

Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:29]:
```suggestion
MPP 模式下，TiFlash 可以......。当没有生成 MPP 执行计划的时候，可以尝试强制开启 MPP：
```

File: tiflash/tune-tiflash-performance.md (hunk 8 above)
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:70]:
```suggestion

使用 Shuffled Hash Join 时，TiFlash ......，存在 ......（性能问题）。采用 Broadcast Hash Join，可以使 TiFlash 性能 .......。
```

File: tiflash/tune-tiflash-performance.md [hunk 11]
```diff
@@ -1,64 +1,103 @@
...
+- [设置更大的执行并发度](#设置更大的执行并发度)
+- [设置细粒度 Shuffle 参数](#设置细粒度-shuffle-参数)
 
-    ```sql
-    set @@tidb_distsql_scan_concurrency = 80;
-    ```
+### 强制开启 MPP 模式
 
-2. 开启 Super batch 功能：
+MPP 执行计划可以充分利用分布式计算资源，从而显著提高批量数据查询的效率。当查询没有生成 MPP 执行计划的时候，用户可以尝试强制开启 MPP：
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:29]:
```suggestion
MPP 执行计划可以充分利用分布式计算资源，从而显著提高批量数据查询的效率。当查询没有生成 MPP 执行计划的时候，你可以强制开启 MPP：
```

File: tiflash/tune-tiflash-performance.md [hunk 12]
```diff
@@ -1,64 +1,103 @@
...
 
-    [`tidb_allow_batch_cop`](/system-variables.md#tidb_allow_batch_cop-从-v40-版本开始引入) 变量用来设置从 TiFlash 读取时，是否把 Region 的请求进行合并。当查询中涉及的 Region 数量比较大，可以尝试设置该变量为 `1`（对带 `aggregation` 下推到 TiFlash Coprocessor 的请求生效），或设置该变量为 `2`（对全部下推到 TiFlash Coprocessor 请求生效）。
+[`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入) 变量用于控制是否忽略优化器代价估算，强制使用 TiFlash 的 MPP 模式执行查询。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:31]:
```suggestion
[`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-从-v51-版本开始引入) 变量用于控制是否忽略优化器代价估算，强制使用 TiFlash 的 MPP 模式执行查询。要开启 MPP 模式查询，执行如下命令：
```

File: tiflash/tune-tiflash-performance.md [hunk 13]
```diff
@@ -1,64 +1,103 @@
...
+set @@tidb_enforce_mpp = 1;
+```
 
-    ```sql
-    set @@tidb_allow_batch_cop = 1;
-    ```
+### 聚合下推 `Join` 或 `Union`
 
-3. 尝试开启聚合推过 `Join` / `Union` 等 TiDB 算子的优化：
+将聚合操作下推到 `Join` 或 `Union` 之前执行，有可能能显著减少 `Join` 或 `Union` 需要处理的数据量，从而提升性能。尝试开启聚合推过 `Join` / `Union` 等 TiDB 算子的优化：
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:39]:
```suggestion
将聚合操作下推到 `Join` 或 `Union` 之前执行，有可能能显著减少 `Join` 或 `Union` 需要处理的数据量，从而提升性能。
```

File: tiflash/tune-tiflash-performance.md [hunk 14]
```diff
@@ -1,64 +1,103 @@
...
 
-    [`tidb_opt_agg_push_down`](/system-variables.md#tidb_opt_agg_push_down) 变量用来设置优化器是否执行聚合函数下推到 Join 之前的优化操作。当查询中聚合操作执行很慢时，可以尝试设置该变量为 1。
+[`tidb_opt_agg_push_down`](/system-variables.md#tidb_opt_agg_push_down) 变量用来设置优化器是否执行聚合函数下推到 Join 之前的优化操作。当查询中聚合操作执行很慢时，可以尝试设置该变量为 1。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:41]:
```suggestion
[`tidb_opt_agg_push_down`](/system-variables.md#tidb_opt_agg_push_down) 变量用来设置优化器是否执行聚合函数下推到 Join 之前的优化操作。当查询中聚合操作执行很慢时，可以尝试设置该变量为 `ON`。
```

File: tiflash/tune-tiflash-performance.md [hunk 15]
```diff
@@ -1,64 +1,103 @@
...
 
-    {{< copyable "sql" >}}
+```sql
+set @@tidb_opt_agg_push_down = 1;
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:44]:
```suggestion
set @@tidb_opt_agg_push_down = ON;
```

File: tiflash/tune-tiflash-performance.md [hunk 16]
```diff
@@ -1,64 +1,103 @@
...
+
+[`tidb_max_tiflash_threads`](/system-variables.md#tidb_max_tiflash_threads-从-v610-版本开始引入)，单位为 bytes。用来设置 TiFlash 中 request 执行的最大并发度。
+
+```sql
+set @@tidb_max_tiflash_threads = 20;
+```
+
+### 设置细粒度 Shuffle 参数
+
+细粒度 Shuffle 可以通过参数增加执行窗口函数执行的并法度，让函数执行占用更多系统资源，从而提升查询性能。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:97]:
```suggestion
细粒度 Shuffle 可以通过参数增加窗口函数执行的并法度，让函数执行占用更多系统资源，从而提升查询性能。
```

**Review by shichun-0415 — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
LGTM

File: tiflash/tune-tiflash-performance.md [hunk 17]
```diff
@@ -1,64 +1,103 @@
...
     ```sql
-    ALTER TABLE employees COMPACT PARTITION pNorth, pEast TIFLASH REPLICA;
-    ```
\ No newline at end of file
+    set @@tidb_broadcast_join_threshold_count = 100000;
+    ```
+
+### 设置更大的执行并发度
+
+设置更大的执行并法度，可以让 TiFlash 占用更多系统 CPU 资源，从而提升查询性能。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:87]:
```suggestion
设置更大的执行并发度，可以让 TiFlash 占用更多系统 CPU 资源，从而提升查询性能。
```

File: tiflash/tune-tiflash-performance.md [hunk 18]
```diff
@@ -1,64 +1,103 @@
...
+
+[`tidb_max_tiflash_threads`](/system-variables.md#tidb_max_tiflash_threads-从-v610-版本开始引入)，单位为 bytes。用来设置 TiFlash 中 request 执行的最大并发度。
+
+```sql
+set @@tidb_max_tiflash_threads = 20;
+```
+
+### 设置细粒度 Shuffle 参数
+
+细粒度 Shuffle 可以通过参数增加窗口函数执行的并法度，让函数执行占用更多系统资源，从而提升查询性能。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:97]:
```suggestion
细粒度 Shuffle 可以通过参数增加窗口函数执行的并发度，让函数执行占用更多系统资源，从而提升查询性能。
```

**Pull request merged**